h/services/jwt.py

```diff
@@ -1,35 +1,77 @@
-import datetime
+from collections.abc import Iterable
+from dataclasses import asdict
+from datetime import datetime, timedelta
+from enum import StrEnum
 from functools import lru_cache
-from typing import Any
+from typing import Any, TypeVar
 
 import jwt
 from jwt import PyJWKClient
-from jwt.exceptions import PyJWTError
+from jwt.exceptions import InvalidTokenError, PyJWTError
 
 JWK_CLIENT_TIMEOUT = 10
 
 
-class TokenValidationError(Exception):
+class JWTDecodeError(Exception):
     """Decoding or validating a JWT failed."""
 
 
+class JWTPayloadError(JWTDecodeError):
+    """A JWT decoded successfully but the payload wasn't what we expected."""
+
+
+class JWTAudiences(StrEnum):
```

Comment:
Single, centralised `JWTAudiences` and `JWTIssuers` enums are used for that in the future when we have multiple disparate parts of the code calling `JWTService` to encode JWTs we can be sure that none of them are unwittingly duplicating audience or issuer strings.